src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,68 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        // Try to resolve implementation for the given interface
+        Class<?> implClass = findImplementation(interfaceClazz);
+
+        // Check if class is annotated with @Component
+        if (!implClass.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Class " + implClass.getName()
+                    + " is not marked as @Component");
```

Comment:
When throwing an exception add an informative message to it. Also, don't forget to add an exception you're catching in `catch` block to the `RuntimeException` object you throw. This will provide more context in the error message and make debugging easier.